### PR TITLE
Devel environment

### PR DESCRIPTION
**Pull request by zergin**
As I'd like to push doThings forward a bit to scratch some of my itches here is a PR for development environment. 

This adds docker-compose support creating three containers: nginx, php and a gulp builder. Each container adheres to "single-process" mantra. The PHP container is run as unprivileged user. Also composer dependencies are updated so that there are no fatal errors (and deprecation warnings) during install. 

**NOTE:** Entire environment is wired for development (app_dev.php Symfony front controller), host mounted volumes with sources, gulp builder watching. 

To run:

```
docker-compose up
```

To stop and clean:

```
docker-compose down
```

Docker-compose 1.7+ is required (and docker-engine 1.11+) because of v2 configuration file.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,8 +14,8 @@
         "twig/extensions": "~1.0",
         "symfony/swiftmailer-bundle": "~2.3",
         "symfony/monolog-bundle": "~2.4",
-        "sensio/distribution-bundle": "^3.0.12",
-        "sensio/framework-extra-bundle": "^3.0.2",
+        "sensio/distribution-bundle": "^3.0",
+        "sensio/framework-extra-bundle": "^3.0",
```
Comment (DavidBadura) [inline composer.json:18]: why did you change that?

File: bin/runner.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -x
+
+npm install -g gulp-cli
```
Comment (DavidBadura) [inline bin/runner.sh]: you don't need gulp-cli

**Comment by DavidBadura** [comment]
otherwise :+1:

File: composer.json (hunk 1 above)
Comment (zergin) [inline composer.json:18]: Because of current composer that requires distribution-bundle afaik > 3.0.18 (event classes changes). I had post-install script errors w/o this change.

Do you see any problem with locking to 3.0.x? It seems to be working fine for me in preliminary tests.

**Comment by zergin** [comment]
I've removed adding gulp-cli (a leftover from running gulp directly before I saw "scripts" definition in package.json).

I also changed Dockerfile a bit not to include vendors while building. As this is purely development image all requirements must be installed after host volume is mounted anyway (call to `composer install` is done on every container start).

File: bin/runner.sh (hunk 2 above)
Comment (zergin) [inline bin/runner.sh]: removed.

**Comment by DavidBadura** [comment]
thank you :smiley:
